server/protocols/protocols-imap4/src/main/java/org/apache/james/imapserver/netty/IMAPServer.java

```diff
@@ -72,19 +74,12 @@
     public final static int DEFAULT_TIMEOUT = 30 * 60; // default timeout is 30 seconds
     public final static int DEFAULT_LITERAL_SIZE_LIMIT = 0;
 
-    @Inject
-    public void setImapProcessor(ImapProcessor processor) {
+    public IMAPServer(ImapDecoder decoder, ImapEncoder encoder, ImapProcessor processor, Metric imapConnectionsMetric, Metric imapCommandsMetric) {
```

Comment:
I feel that having one service by metric will be a little too much : we risk to have to modify many constructors each time we want to had a metric. WDYT?